packages/transaction-controller/src/utils/transaction-type.ts

```diff
@@ -27,37 +27,48 @@ export async function determineTransactionType(
   ethQuery: EthQuery,
 ): Promise<InferTransactionTypeResult> {
   const { data, to } = txParams;
-  const name = parseStandardTokenTransactionData(data)?.name;
 
   if (data && !to) {
     return { type: TransactionType.deployContract, getCodeResponse: undefined };
   }
 
-  const { contractCode: resultCode, isContractAddress } =
+  const { contractCode: getCodeResponse, isContractAddress } =
     await readAddressAsContract(ethQuery, to);
 
   if (!isContractAddress) {
-    return { type: TransactionType.simpleSend, getCodeResponse: resultCode };
+    return { type: TransactionType.simpleSend, getCodeResponse };
   }
 
   const hasValue = txParams.value && Number(txParams.value) !== 0;
```

Comment:
```suggestion
  const hasValue = txParams.value !== undefined && Number(txParams.value) !== 0;
```

Nit: The type of `hasValue` currently evaluates to `boolean | "" | undefined`. This change fixes it to `boolean`. 

But this wasn't a change introduced in this PR so it doesn't have to be fixed here.

Reply:
Nice.